controllers/pkg/reconcilers/repository/reconciler_test.go

```diff
@@ -196,3 +202,77 @@ func TestDeleteRepo(t *testing.T) {
 		})
 	}
 }
+
+func Test_reconciler_upsertRepo(t *testing.T) {
```

Comment:
This method writes tests for the `upsertRepo` method (already written at https://github.com/nephio-project/nephio/blob/main/controllers/pkg/reconcilers/repository/reconciler_test.go#L65) and demonstrate how this can be written using mockery and testify.